doc/source/xgboost-ray.rst

```diff
@@ -0,0 +1,135 @@
+.. _xgboost-ray:
+
+XGBoost on Ray
+==============
+
+This library adds a new backend for XGBoost utilizing Ray.
+
+Please note that this is an early version and both the API and
+the behavior can change without prior notice.
+
+Installation
+------------
+
+You can install ``xgboost_ray`` like this:
+
+.. code-block:: bash
+
+    git clone https://github.com/ray-project/xgboost_ray.git
+    cd xgboost_ray
+    pip install -e .
+
+
+Usage
+-----
+
+``xgboost_ray`` provides a drop-in replacement for XGBoost's ``train``
+function. To pass data, instead of using ``xgb.DMatrix`` you will
+have to use ``xgboost_ray.RayDMatrix``.
+
+Here is a simplified example:
+
+.. code-block:: python
+
+    from xgboost_ray import RayDMatrix, train
```

Comment:
Should we create a python script and add it to the CI to make sure we pick up any API changes?